### PR TITLE
14/create band

### PR DESCRIPTION
**Pull request by Matthew-Steen**
## Description
Changes made for this branch

## Checklist
- [x] All manual tests passing
- [ ] All unit tests passing 
- [ ] All UI tests passing
- [ ] Code reviewed by peer:\
&nbsp; - *No redundant whitespace*\
&nbsp; - *Indentation correct*\
&nbsp; - *Naming conventions followed*\
&nbsp; - *Comments added (where needed)*\
&nbsp; - *No commented out redundant code*\
&nbsp; - *Branch name correct*
- [ ] Code reviewed by TA

### CODE REVIEW

File: Angular/src/app/add-band/add-band.component.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import { Band } from "../model/band";
+import { DataService } from "../_services/data.service";
+import { Observable } from 'rxjs';
+
+const DESCRIPTION_MAX_LEN = 16777215
```
Comment (matthewdoreilly) [inline Angular/src/app/add-band/add-band.component.ts]: Why so long?!

File: Express-API/band.js [hunk 2]
```diff
@@ -1,37 +1,47 @@
-const getBandQuery = "SELECT * FROM viewBandData WHERE band_id = ?"
-const allBandsQuery = "SELECT * FROM band"
+const getBandQuery = "SELECT * FROM viewBandData WHERE band_id = ? order by school_id ASC, priority ASC"
+const allBandsQuery = "SELECT * FROM band order by school_id ASC, priority ASC"
```
Comment (matthewdoreilly) [inline Express-API/band.js]: Where is the SQL change for this `priority` column?

File: Express-API/band.js [hunk 3]
```diff
@@ -1,37 +1,47 @@
...
-                        params.responsibilities
-                    ]).then(result => {
+    addBand(params, db) {
+        return new Promise((resolve, reject) => {
+        try {
+            // Insert description data into 'band_description' table
+            db.query(addBandDescriptionQuery,[params.commercialAwareness,params.communicatingAndTeamwork,
+            params.innovationAndContinuousImprovement,params.customerFocus,
+            params.developingYourselfAndOthers,params.planningAndOrganisation,
+            params.jobSpecificKnowledge]).then(rows => {
```
Comment (matthewdoreilly) [inline Express-API/band.js]: Indent this to make it more readable, like:

```
try {
     // Insert description data into 'band_description' table
     db.query(addBandDescriptionQuery, [
            params.commercialAwareness, params.communicatingAndTeamwork,
            params.innovationAndContinuousImprovement, params.customerFocus,
            params.developingYourselfAndOthers, params.planningAndOrganisation,
            params.jobSpecificKnowledge
      ]).then(rows => {
            ...
```

File: Angular/src/app/admin-table/admin-table.component.html [hunk 4]
```diff
@@ -19,19 +19,22 @@
               <th class='job-family'>
                 <div>
                   <button class='btn btn-primary' id='addCapability' (click)="openCapabilityModal()">+ Add Capability</button>
+                  <!-- <button class='btn btn-primary' id='addBand' (click)="openAddBandModal()">+ Add Band</button> -->
```
Comment (jamescrone) [inline Angular/src/app/admin-table/admin-table.component.html]: is there a reason why this line is commented out? Is it required

File: Angular/src/app/model/band.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+export class Band {
```
Comment (matthewdoreilly) [inline Angular/src/app/model/band.ts]: Think you can make this an interface, `export interface Band`

File: Angular/src/app/add-band/add-band.component.ts (hunk 1 above)
Comment (matthewdoreilly) [inline Angular/src/app/add-band/add-band.component.ts]: Think you may have missed my comment above here

**Pull request merged**